From 7191fa084bc8496b932d2e29d978335dcb7f0ee6 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 11 Mar 2019 14:04:23 +0100
Subject: [PATCH] arm64: dt: prepare fsl-imx8dx[p]-tqma8dx[p]-mba8xx

simply based on existing tree and fragments for fsl-imx8qxp-tqma8qx-mba8qx

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 ++
 .../freescale/fsl-imx8dx-tqma8dx-mba8xx.dts   | 23 +++++++++++++++++++
 .../dts/freescale/fsl-imx8dx-tqma8dx.dtsi     | 21 +++++++++++++++++
 .../freescale/fsl-imx8dxp-tqma8dxp-mba8xx.dts | 23 +++++++++++++++++++
 .../dts/freescale/fsl-imx8dxp-tqma8dxp.dtsi   | 21 +++++++++++++++++
 5 files changed, 90 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8dx-mba8xx.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8dx.dtsi
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8dxp-mba8xx.dts
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8dxp.dtsi

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index aff1c5eccb07..9e88219754fb 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -89,7 +89,9 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-a0.dtb \
 				  fsl-imx8qxp-ddr3l-val.dtb \
 				  fsl-imx8dx-lpddr4-arm2.dtb \
+				  fsl-imx8dx-tqma8dx-mba8xx.dtb \
 				  fsl-imx8dxp-lpddr4-arm2.dtb \
+				  fsl-imx8dxp-tqma8dxp-mba8xx.dtb \
 				  fsl-imx8qxp-tqma8qx-mba8qx.dtb \
 				  fsl-imx8qxp-tqma8qx-mba8qx-lvds-tm070jvhg33.dtb \
 				  fsl-imx8qxp-tqma8qxs-mb-smarc-2.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8dx-mba8xx.dts b/arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8dx-mba8xx.dts
new file mode 100644
index 000000000000..ef808f56beca
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8dx-mba8xx.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "fsl-imx8dx-tqma8dx.dtsi"
+#include "fsl-imx8qxp-tqma8qx-mba8qx.dtsi"
+
+/ {
+	model = "TQ Systems i.MX8DX TQMa8DX on MBa8xX";
+	compatible = "tq,tqma8dx-mba8xx", "tq,tqma8dx", "fsl,imx8dx";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8dx.dtsi
new file mode 100644
index 000000000000..8ca09331603b
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx-tqma8dx.dtsi
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8dx.dtsi"
+#include "tqma8xx-common.dtsi"
+
+/ {
+	model = "TQ Systems i.MX8DX TQMa8DX";
+	compatible = "tq,tqma8dx", "fsl,imx8dx";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8dxp-mba8xx.dts b/arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8dxp-mba8xx.dts
new file mode 100644
index 000000000000..483078d9318c
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8dxp-mba8xx.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "fsl-imx8dxp-tqma8dxp.dtsi"
+#include "fsl-imx8qxp-tqma8qx-mba8qx.dtsi"
+
+/ {
+	model = "TQ Systems i.MX8DXP TQMa8DXP on MBa8xX";
+	compatible = "tq,tqma8dxp-mba8xx", "tq,tqma8dxp", "fsl,imx8dxp";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8dxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8dxp.dtsi
new file mode 100644
index 000000000000..f9f284cb0c98
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dxp-tqma8dxp.dtsi
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2019 TQ Systems GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8dxp.dtsi"
+#include "tqma8xx-common.dtsi"
+
+/ {
+	model = "TQ Systems i.MX8DXP TQMa8DXP";
+	compatible = "tq,tqma8dxp", "fsl,imx8dxp";
+};
